src/Common/tests/System/Net/Http/LoopbackProxyServer.cs

```diff
@@ -218,41 +240,53 @@ private async Task ProcessConnectMethod(NetworkStream clientStream, string remot
             {
                 try
                 {
-                    byte[] buffer = new byte[8000];
-                    int bytesRead;
-                    while ((bytesRead = await clientStream.ReadAsync(buffer, 0, buffer.Length).ConfigureAwait(false)) > 0)
-                    {
-                        await serverStream.WriteAsync(buffer, 0, bytesRead).ConfigureAwait(false);
-                    }
-                    serverStream.Flush();
+                    await clientStream.CopyToAsync(serverStream).ConfigureAwait(false);
                     serverSocket.Shutdown(SocketShutdown.Send);
                 }
-                catch (IOException)
+                catch (IOException ex) when (IsRudeDisconnect(ex))
                 {
                     // Ignore rude disconnects from either side.
+                    // Close sockets to abort the other task.
+                    clientStream.Close();
+                    serverStream.Close();
+                }
+                catch
+                {
+                    clientStream.Close();
+                    serverStream.Close();
+                    throw;
```

Comment:
Seems like this would be cleaner as:
```C#
catch (Exception ex)
{
    clientStream.Close();
    serverStream.Close();
    if (!IsRudeDisconnect(ex))
    {
        throw;
    }
}
```
rather than duplicating the code, having the extra catch block, etc.

Reply:
Refactored this to remove duplication.